### DISCUSSION TITLE
How to handle duplicate pings in validation

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1527] [General]
I have a topology in which ICMP packets get duplicated for some reason. Validate returns "success"

Should we fail the ping test in case "DUP!" is in the output? Issue a warning? ...

**Comment by jbemmel** [comment]
Side note: I finally discovered why @ssasso prefers virtual networks on OS10: https://www.reddit.com/r/networking/comments/1494owm/os10_vlt_vrrp_causing_packets_to_double/

On Dell OS10, regular vlan interfaces with VRRP cause ping packet duplication (even without VLT)

**Reply by ipspace** [reply]
> On Dell OS10, regular vlan interfaces with VRRP cause ping packet duplication (even without VLT)

I guess that answers my rhetorical question ;))

**Comment by DanPartelly** [comment]
Now you have an interesting problem at your hands. Figure out why.  It might be harmless, it might be a bug in the device. 

But anyway, I would try to figure it out. You can always share those test topologies which yiled interesting cases. Of course if there is no proprietary stuff modeled  with it. 

> I have a topology in which ICMP packets get duplicated for some reason. Validate returns "success"
> 
> Should we fail the ping test in case "DUP!" is in the output? Issue a warning? ...

**Comment by ipspace** [comment]
> I have a topology in which ICMP packets get duplicated for some reason. Validate returns "success"

Would perchance an OS10 with a first-hop gateway be involved? ;) Or a bridging loop?

On a more serious note, the last time I've seen something like that was on a Catalyst 6500 running MPLS over GRE. Virtual network devices are unfortunately just a (decent or abhorrent) approximation of the physical devices.

> Should we fail the ping test in case "DUP!" is in the output? Issue a warning? ...

How often do you think that happens (as in "in how many scenarios" not "every time you do ping"), and is it worth caring about it? You can always write your own code to evaluate the ping results, either as a plugin or as an inline Python expression.
